examples/thermometer/src/main.rs

```diff
@@ -0,0 +1,160 @@
+#![no_main]
+#![no_std]
+
+mod i2c_bus;
+mod pins;
+mod sensors;
+
+use ariel_os::{
+    debug::log::{error, info},
+    sensors::{
+        Category, Label, MeasurementUnit, REGISTRY, Reading as _,
+        sensor::{ReadingChannel, Sample},
+    },
+    time::Timer,
+};
+
+use stm32_lcd_driver::{Digit, Lcd};
+
+#[ariel_os::task(autostart, peripherals)]
+async fn main(peripherals: pins::Peripherals) {
+    let pins::Peripherals {
+        lcd: lcd_peris,
+        pins: pin_peris,
+        i2c: i2c_peris,
+    } = peripherals;
+    i2c_bus::init(i2c_peris);
+    sensors::init().await;
+
+    info!("Will print the readings of temperature sensor on the LCD Screen");
+
+    let mut lcd = Lcd::new(lcd_peris.lcd, pin_peris.into_pins());
+    lcd.initialize().await;
+
+    loop {
+        // Trigger measurements for each sensor driver in parallel.
+        let Some(sensor) = REGISTRY
+            .sensors()
+            .find(|s| s.categories().contains(&Category::Temperature)) else {
+            info!("There aren't any registered temperature sensors");
+            break;
+        };
+
+        if let Err(err) = sensor.trigger_measurement() {
+            error!("Error when triggering a measurement: {}", err);
+            Timer::after_secs(2).await;
+            continue;
+        }
+        let reading = sensor.wait_for_reading().await;
+
+        match reading {
+            Ok(samples) => {
+                for (reading_channel, sample) in
+                    samples.samples().filter(|(reading_channel, _)| {
+                        matches!(reading_channel.label(), Label::Temperature)
```

Comment:
This is functionally equivalent, but otherwise it looks like `Label` doesn't implement `PartialEq`:
```suggestion
                        reading_channel.label() == Label::Temperature
```